From 0eabaeb03036d9a9381e3bbd69ae4bf5af3b5f01 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 04 Nov 2020 20:15:47 +0200
Subject: [PATCH] [SW-12755] security: introduce security.h header

As a preparation for dynamic security introduce the generic security.h
header file and move the init declaration.

Change-Id: Ib95b62b8d60d8943f9b415791461ddd1f7d2913d
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/include/preboot.h b/include/preboot.h
index 72cd473..9b81139 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -110,7 +110,6 @@
 void preboot_delay(void);
 void preboot_putc_virt_uart(const char c);
 void preboot_sync_btl(void);
-void preboot_security_init(void);
 void preboot_led_on(void);
 
 void ppboot_cp_preboot(u64 loadaddr, u32 offset, u32 size, char *name);
diff --git a/include/security.h b/include/security.h
new file mode 100644
index 0000000..7c9a567
--- /dev/null
+++ b/include/security.h
@@ -0,0 +1,12 @@
+/* SPDX-License-Identifier: GPL-2.0+ */
+/*
+ * Copyright 2016-2020 HabanaLabs, Ltd.
+ * All Rights Reserved.
+ */
+
+#ifndef _HL_SECURITY_H_
+#define _HL_SECURITY_H_
+
+void preboot_security_init(void);
+
+#endif /* _HL_SECURITY_H_ */
diff --git a/preboot/preboot.c b/preboot/preboot.c
index 2dce446..c2d722b 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -13,6 +13,7 @@
 #include <hl_fw_if.h>
 #include <hl_boot_if.h>
 #include <boot_if.h>
+#include <security.h>
 
 #include "common.h"
 #include "uart.h"
